string2.py

```diff
@@ -42,9 +50,16 @@ def not_bad(s):
 # e.g. 'abcde', the front half is 'abc', the back half 'de'.
 # Given 2 strings, a and b, return a string of the form
 #  a-front + b-front + a-back + b-back
+def divide_string( s ):
+    if len(s) % 2 == 0:
+        return [ s[:len(s)/2], s[len(s)/2:] ]
+    else:
+        return [ s[:len(s)//2 + 1 ], s[len(s)//2 + 1 :] ]
+
 def front_back(a, b):
-    """Your code goes here.  Edit this docstring."""
-    return
+    [ s1, s3 ] = divide_string(a)
```

Comment:
You should not need the square brackets around these, or around the values returned from divide_string.
Also, I've been encouraging folks to not insert too much embedded math into indexing [] brackets.  It becomes difficult to read very quickly. Better:
```
midpoint = len(s) / 2
if midpoint % 2:
    midpoint +=1
return s[:midpoint], s[midpoint:]
```